src/result.test.ts

```diff
@@ -1198,6 +1198,147 @@ describe("Result", () => {
 			});
 		});
 
+		describe("mapError", () => {
```

Comment:
I see that you use `CustomError` a lot in the added test cases. At first glance there is nothing particularly wrong with this, although type-wise it tend to 'slip through the cracks' here and there. Are you ok with using the `ErrorA` and `ErrorB` classes to proof things like transformations of errors, etc? As bonus you only have to match against the constructor of the error.